Add section clarifying pip version support (#11556)
pypa/pip: docs/html/development/release-process.rst

  ``pip._internal.utils.deprecation.deprecated``. The function is not a part of
  pip's public API.

> .. _`Supported Versions`:

Reviewer: Let's not add this intersphinx reference -- nothing in our docs use this.

Author: done!

> .. _`Supported Versions`:
> 
> Supported Versions
> ==================

Author: ```suggestion
Supported Versions
----------------
```

> The latest major version of the pip is the only supported version, previous

Reviewer 2: ```suggestion
The latest version of the pip is the only supported version, previous
```

> major versions should be considered unsupported. Folks are encouraged to make

Reviewer 2: ```suggestion
versions should be considered unsupported. Users are encouraged to make
```

> regular updates to their version of pip for best performance.

Reviewer 2: ```suggestion
regular updates to their version of pip in order to remain supported.
```
